core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -312,25 +312,30 @@ public int hashCode() {
         return nodeId.hashCode();
     }
 
+    private String toString;
```

Comment:
You're not convinced of what? Not convinced that we should eagerly evaluate it? I'm fine with that. Out of curiosity, did you look at the places in `ZenDiscovery` where this is used? Note that it's also used in `IndicesClusterStateService#onRecoveryDone` or `IndicesClusterStateService#applyInitializingShard` or when a node is added to the cluster and so used in `ClusterService#runTasksForExecutor`? Maybe some other places too but they are tricky to find (since it's usually invoked indirectly by having it implicitly passed into `StringBuilder#append(Object)`).